framework/include/variables/MooseVariableBase.h

```diff
@@ -178,6 +178,11 @@ class MooseVariableBase : public MooseObject,
    */
   bool isArray() const { return _is_array; }
 
+  /**
+   * @return whether this variable lives on lower dimensional blocks
+   */
+  bool isLowerD() const { return _is_lower_d; }
```

Comment:
can a variable be both lowerD and normal D?

Reply:
no, that would make evaluation on a boundary face ambiguous